### DISCUSSION TITLE
Total Requests Limiting (regardless of time-window)

### DISCUSSION THREAD

**Discussion by fenix-hub** [Kong/kong#9742] [Help] [answered]
Hi everyone.
I'm aware of the Rate Limiting and Advanced Rate Limiting plugins, to limit the rate of requests, but I wonder if there is a plugin (or a setting) to actually limit client requests based on the *total* amount of requests per-user (/ip/path/etc..).
Instead of configuring "how many requests per _{time window}_ can be executed by user by header" on a single Service, I'd like to be able to configure "how many requests can be executed by a user by header" in total, so letting the plugin count how many requests have been processed and stop when the limit is reached, regardless of the time-window.
Can be Rate Limiting plugins used for this? Is there a plugin to do this? Or do I need to develop my own plugin?
Thanks

**Comment by fenix-hub** [comment]
Any update?

**Comment by samugi** [comment, marked as answer]
@fenix-hub would a very large window_size (i.e. 100 years or more) suit your use case? We could think of the total amount of requests from a user as the requests sent during a virtually infinite time window by such user. Setting `window_size=0` might also work.

**Reply by fenix-hub** [reply]
Yes, using a very large window_size in years could be a *temporary* solution, even though it could be improved.
Being able to set the window_size=0 would be good, even though you won't still be able to set the _maximum_ amount of requests, since in the plugin (at least the community one) I need to specify one of the window sizes (seconds, minutes, etc.)
